### PR TITLE
Initial refactor of the Demo

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR introduces the first set of refactorings to the JxBrowser Demo.

Some of the TODOs are left in the code for future improvements.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@vladimir-ikryanov, PTAL.

File: .idea/codeStyles/codeStyleConfig.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<component name="ProjectCodeStyleConfiguration">
+  <state>
+    <option name="PREFERRED_PROJECT_CODE_STYLE" value="Default (1)" />
```
Comment (vladimir-ikryanov) [inline .idea/codeStyles/codeStyleConfig.xml]: Maybe we should rename the Code Style configuration to something more readable than `Default (1)`?

File: demo/src/main/java/com/teamdev/jxbrowser/demo/Application.java [hunk 2]
```diff
@@ -21,29 +21,30 @@
...
 import java.awt.event.WindowEvent;
 
-public class JxBrowserDemo {
+import static com.teamdev.jxbrowser.demo.resources.Resources.loadIcon;
 
-    private static void initEnvironment() throws Exception {
-        System.setProperty("apple.laf.useScreenMenuBar", "true");
-        System.setProperty("com.apple.mrj.application.apple.menu.about.name", "JxBrowser Demo");
-        UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+final public class Application {
```
Comment (vladimir-ikryanov) [inline demo/src/main/java/com/teamdev/jxbrowser/demo/Application.java]: Should be `public final class...` according to https://docs.oracle.com/javase/specs/jls/se8/html/jls-8.html#jls-8.1.1

File: demo/src/main/java/com/teamdev/jxbrowser/demo/JsConsole.java [hunk 3]
```diff
@@ -142,12 +124,25 @@ private void updateConsoleOutput(String script, String executionResult) {
...
         console.append(script);
-        console.append(NEW_LINE);
+        newLine();
+    }
+
+    private void newLine() {
+        console.append("\n");
+    }
+
+    static class Event {
```
Comment (vladimir-ikryanov) [inline demo/src/main/java/com/teamdev/jxbrowser/demo/JsConsole.java]: Usually the constant holder classes are `finale`.

File: demo/src/main/java/com/teamdev/jxbrowser/demo/Menu.java [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+import java.awt.event.MouseEvent;
+import java.io.File;
+
+import static javax.swing.JOptionPane.INFORMATION_MESSAGE;
+import static javax.swing.JOptionPane.showMessageDialog;
+
+/**
+ * Application menu shown from a drop-down button.
+ *
+ * @author unascribed
```
Comment (vladimir-ikryanov) [inline demo/src/main/java/com/teamdev/jxbrowser/demo/Menu.java]: I don't think it's a good idea to use "unascribed" as the argument to `@author`. This is our product. We do know the authors :)

In general, I would avoid using the `@author` tab at all. I added an appropriate [chapter](https://github.com/TeamDev-Ltd/JxBrowser/wiki/Java-Style-Guide#74-no-author-tags) to our [Java Style Guide](https://github.com/TeamDev-Ltd/JxBrowser/wiki/Java-Style-Guide).

**Pull request merged**